test/request.test.ts

```diff
@@ -878,4 +880,36 @@ x//0u+zd/R/QRUzLOw4N72/Hu+UG6MNt5iDZFCtapRaKt6OvSBwy8w==
       );
     });
   });
+
+  it("validate request with readstream data", () => {
+    const size = fs.statSync(__filename).size;
+    const mock = fetchMock
+      .sandbox()
+      .post(
+        "https://api.github.com/repos/octokit-fixture-org/release-assets/releases/v1.0.0/assets",
+        {
+          status: 200,
+        }
+      );
+
+    return request("POST /repos/{owner}/{repo}/releases/{release_id}/assets", {
+      owner: "octokit-fixture-org",
+      repo: "release-assets",
+      release_id: "v1.0.0",
+      request: {
+        fetch: mock,
+      },
+      headers: {
+        "content-type": "text/json",
+        "content-length": size,
+      },
+      data: fs.createReadStream(__filename),
+      name: "test-upload.txt",
+      label: "test",
+    }).then((response) => {
+      expect(response.status).toEqual(200);
+      expect(mock.lastOptions()?.body instanceof stream.Readable).toBe(true);
```

Comment:
```suggestion
      expect(mock.lastOptions()?.body).toBeInstanceOf(stream.Readable);
```

Reply:
other than what @wolfy1339 said here this PR looks great 👍🏼

Reply:
Committed this suggestion! Thanks!